test/core/config_test.py

```diff
@@ -100,7 +100,7 @@ def test_config_get_validators(temp_dir):
         }
```

Comment:
There's no tests for the new validation download

Reply:
This is going to be tricky as I'll need to mock the github api, but I can give it a go

Reply:
Usually I want self-contained tests, but for this case my initial thought would be to just set up a github repo to test against. Mocking all that stuff seems like too much work for too little gain over just relying on an external resource. Is there a reason I'm not seeing that that's not doable or a bad idea?

Reply:
... AND ANOTHER THING... mocking external dependencies is an anti-pattern anyway since if the external APIs change your tests will continue to pass but your actual deployments will fail

Reply:
I don't see a test for a few cases:

* No url or repo supplied
* the no releases case
* the no assets case
* parser errors

Reply:
For parser errors, they're just the same error one gets from the URL flow. I don't see good reason to go through the process of retesting that part. The others I can add cases for. Perhaps excluding the "no releases" case, thats something I'm less comfortable with testing without a mock because it's really easy to break that test if someone ends up creating a release in whatever repo is targeted.

Reply:
> For parser errors, they're just the same error one gets from the URL flow. 

maybe I'm misunderstanding again, but it looks like there's one branch for `config_asset==True` and one for `False`, and only the latter is tested?

Reply:
> Perhaps excluding the "no releases" case, thats something I'm less comfortable with testing without a mock because it's really easy to break that test if someone ends up creating a release in whatever repo is targeted.

make a new repo and archive it, and note in the readme it's for X test in Y repo, don't touch it?

Reply:
Oh, I missed a case for where the repo is provided but no asset filename

Reply:
coverage info would make this a lot simpler :P

Reply:
There's this org for test repos btw https://github.com/kbasetest

Reply:
Actually the "no assets" case seems to never happen at the moment on GitHub as all releases include an asset file of the zipped source.

Reply:
Unfortunately I have no privs for https://github.com/kbasetest

Reply:
> Actually the "no assets" case seems to never happen at the moment on GitHub as all releases include an asset file of the zipped source.

I'd ditch that check and add a comment mentioning that in that case

Reply:
> Unfortunately I have no privs

Invited

Reply:
Added tests for check_bool, no releases, no URL or repo_path, and repo_path but no repo_asset. Unfortunately still have no idea why coverage isnt running on this PR but I dont see it on many other PRs in the repo...

Reply:
All I can say is that it ran on every PR I ever did from very early on, I think it was one of the first things I set up